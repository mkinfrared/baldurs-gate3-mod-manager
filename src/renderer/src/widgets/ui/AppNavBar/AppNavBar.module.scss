.AppNavBar {
  --border-color: var(--primary-2);

  @media (prefers-color-scheme: light) {
    --border-color: var(--primary-3);
  }

  border-right: 1px solid var(--border-color);
  display: flex;
  flex: 1;
  flex-direction: column;
  padding: 1rem 0.5rem;
}
