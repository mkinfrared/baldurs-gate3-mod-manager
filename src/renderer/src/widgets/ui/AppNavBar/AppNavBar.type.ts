export type AppNavBarProps = {
  /**
   * a string that will be applied as a css class to parent element
   */
  className?: string;
};
