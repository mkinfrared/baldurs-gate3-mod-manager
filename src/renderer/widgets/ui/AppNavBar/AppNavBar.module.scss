.AppNavBar {
  display: flex;
  flex: 1;
  flex-direction: column;
  padding: 1rem 0.5rem;
}
