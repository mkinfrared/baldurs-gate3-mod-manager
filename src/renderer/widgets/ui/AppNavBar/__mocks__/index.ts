export { default } from "./AppNavBar.mock";
