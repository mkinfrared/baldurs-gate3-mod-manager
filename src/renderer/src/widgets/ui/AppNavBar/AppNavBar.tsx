import { Link } from "react-router-dom";

import { APP_ROUTES } from "@renderer/routes/AppRoutes/AppRoutes.type";
import bg3Logo from "@renderer/shared/assets/images/BG3-logo.png";
import { classNames } from "@renderer/shared/lib/helpers";

import css from "./AppNavBar.module.scss";
import { AppNavBarProps } from "./AppNavBar.type";

const AppNavBar = ({ className }: AppNavBarProps) => (
  <div className={classNames(css.AppNavBar, className)} data-testid="AppNavBar">
    <Link to={APP_ROUTES.BaldursGate3.path}>
      <img src={bg3Logo} alt="Baldur's Gate 3 Logo" />
    </Link>
  </div>
);

export { AppNavBar };
