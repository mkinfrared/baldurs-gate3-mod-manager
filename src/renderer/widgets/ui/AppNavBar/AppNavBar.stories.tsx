import { Meta, StoryObj } from "@storybook/react";

import { AppNavBar } from "./AppNavBar";
import css from "./Story.module.scss";

type Story = StoryObj<typeof meta>;

const meta = {
  title: "UI/AppNavBar",
  component: AppNavBar,
  tags: ["autodocs"],
} satisfies Meta<typeof AppNavBar>;

const Default: Story = {
  args: {},
  render: (args) => (
    <div className={css.Story}>
      <AppNavBar {...args} />
    </div>
  ),
};

export { Default };

export default meta;
