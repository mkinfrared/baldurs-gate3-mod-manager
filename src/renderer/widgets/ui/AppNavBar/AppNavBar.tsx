import css from "./AppNavBar.module.scss";
import { AppNavBarProps } from "./AppNavBar.type";

const AppNavBar = ({ className }: AppNavBarProps) => {
  const classNames = [css.AppNavBar, className].filter(Boolean).join(" ");

  return (
    <div className={classNames} data-testid="AppNavBar">
      AppNavBar works!!
    </div>
  );
};

export { AppNavBar };
