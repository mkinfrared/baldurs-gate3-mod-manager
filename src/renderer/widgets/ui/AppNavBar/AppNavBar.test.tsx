import { render } from "@testing-library/react";

import { AppNavBar } from "./AppNavBar";

describe("<AppNavBar />", () => {
  const Component = <AppNavBar />;

  it("should be defined", () => {
    expect(AppNavBar).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("AppNavBar");

    expect(element).toBeDefined();
  });
});
