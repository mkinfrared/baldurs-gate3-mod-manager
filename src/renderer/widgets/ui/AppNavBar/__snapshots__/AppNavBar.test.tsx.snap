// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppNavBar /> should match the snapshot 1`] = `
<div>
  <div
    class="AppNavBar"
    data-testid="AppNavBar"
  >
    <a
      href="/baldurs-gate-3"
    >
      <img
        alt="Baldur's Gate 3 Logo"
        src="BG3-logo.png"
      />
    </a>
  </div>
</div>
`;
